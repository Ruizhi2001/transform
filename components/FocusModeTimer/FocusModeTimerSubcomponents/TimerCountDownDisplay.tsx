import React, { useState } from "react";
import { Text, View, StyleSheet, Button } from "react-native";

type Props = {
  timerDate: Date;
};

export const TimerCountDownDisplay: React.FC<Props> = ({ timerDate }) => {
  return (
    <View>
      <Text style={styles.timerCountDownText}>
        {timerDate.getHours().toString().padStart(2, "0")}:
        {timerDate.getMinutes().toString().padStart(2, "0")}:
        {timerDate.getSeconds().toString().padStart(2, "0")}
      </Text>
    </View>
  );
};

const styles = StyleSheet.create({
  timerCountDownText: {
    fontSize: 90,
    fontWeight: "700",
  },
});
