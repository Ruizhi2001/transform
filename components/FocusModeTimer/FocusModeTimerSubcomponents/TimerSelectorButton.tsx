import { View, Text, TouchableOpacity, StyleSheet } from "react-native";
import React from "react";

type Props = {
  duration: number;
  toggleTimer: (duration: number) => void;
};

export const TimerSelectorButton: React.FC<Props> = ({
  duration,
  toggleTimer,
}) => {
  return (
    <View>
      <TouchableOpacity
        style={styles.button}
        onPress={() => toggleTimer(duration)}
      >
        <Text style={styles.text}>{duration}</Text>
      </TouchableOpacity>
    </View>
  );
};

const styles = StyleSheet.create({
  button: {
    alignItems: "center",
    padding: 10,
    margin: 10,
    borderColor: "green",
    borderWidth: 6,
    borderRadius: 10,
  },
  text: {
    color: "black",
  },
});
