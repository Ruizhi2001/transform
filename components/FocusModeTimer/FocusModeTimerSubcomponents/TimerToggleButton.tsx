import React from "react";
import { Pressable, View, StyleSheet, Text } from "react-native";
import { AntDesign } from "@expo/vector-icons";

type Props = {
  isTimerRunning: boolean;
  stopTimer: () => void;
  startTimer: () => void;
};

export const TimerToggleButton: React.FC<Props> = ({
  isTimerRunning,
  startTimer,
  stopTimer,
}) => {
  return (
    <Pressable
      onPress={isTimerRunning ? stopTimer : startTimer}
      style={styles.pressable}
    >
      <View>
        <Text style={styles.text}>
          {isTimerRunning ? "ðŸ˜¢ Give up ðŸ˜¢" : "âš¡TRANSFORM NOW!âš¡"}
        </Text>
      </View>
    </Pressable>
  );
};

const styles = StyleSheet.create({
  text: {
    fontSize: 30,
    fontWeight: "700",
  },
  pressable: {
    borderColor: "gold",
    borderWidth: 3,
    borderRadius: 10,
    alignItems: "center",
  },
});
