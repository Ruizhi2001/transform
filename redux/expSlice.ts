import { createSlice } from "@reduxjs/toolkit";
import type { PayloadAction } from "@reduxjs/toolkit";
import type { RootState } from "./store";

// Define a type for the slice state
interface expState {
  value: number;
}

// Define the initial state using that type
const initialState: expState = {
  value: 0,
};

export const expSlice = createSlice({
  name: "exp",
  // `createSlice` will infer the state type from the `initialState` argument
  initialState,
  reducers: {
    // Use the PayloadAction type to declare the contents of `action.payload`
    setInitalAmount: (state, action: PayloadAction<number>) => {
      state.value = action.payload;
    },
    incrementByAmount: (state, action: PayloadAction<number>) => {
      state.value += action.payload;
    },
  },
});

export const { setInitalAmount, incrementByAmount } = expSlice.actions;

// Other code such as selectors can use the imported `RootState` type
export const selectExp = (state: RootState) => state.exp.value;

export default expSlice.reducer;
