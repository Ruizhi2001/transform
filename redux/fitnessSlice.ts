import { createSlice } from "@reduxjs/toolkit";
import type { PayloadAction } from "@reduxjs/toolkit";
import type { RootState } from "./store";

// Define a type for the slice state
interface fitnessState {
  calories: number;
  completed: string[];
  workout: number;
  minutes: number;
}

// Define the initial state using that type
const initialState: fitnessState = {
  calories: 0,
  completed: [],
  workout: 0,
  minutes: 0,
};

export const fitnessSlice = createSlice({
  name: "fitness",
  // `createSlice` will infer the state type from the `initialState` argument
  initialState,
  reducers: {
    // Use the PayloadAction type to declare the contents of `action.payload`
    resetCompleted: (state) => {
      state.completed = [];
    },
    initCalories: (state, action: PayloadAction<number>) => {
      state.calories = action.payload;
    },
    initWorkout: (state, action: PayloadAction<number>) => {
      state.calories = action.payload;
    },
    initMinutes: (state, action: PayloadAction<number>) => {
      state.calories = action.payload;
    },
    setCalories: (state, action: PayloadAction<number>) => {
      state.calories = action.payload;
    },
    setCompleted: (state, action: PayloadAction<string>) => {
      state.completed = [...state.completed, action.payload];
    },
    setWorkout: (state, action: PayloadAction<number>) => {
      state.workout += action.payload;
    },
    setMinutes: (state, action: PayloadAction<number>) => {
      state.minutes += action.payload;
    },
  },
});

export const {
  resetCompleted,
  setCalories,
  setCompleted,
  setWorkout,
  setMinutes,
  initCalories,
  initMinutes,
  initWorkout,
} = fitnessSlice.actions;

// Other code such as selectors can use the imported `RootState` type
export const selectCalorie = (state: RootState) => state.fitness.calories;
export const selectCompleted = (state: RootState) => state.fitness.completed;
export const selectWorkout = (state: RootState) => state.fitness.workout;
export const selectMinutes = (state: RootState) => state.fitness.minutes;

export default fitnessSlice.reducer;
