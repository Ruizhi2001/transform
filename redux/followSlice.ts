import { createSlice } from "@reduxjs/toolkit";
import type { PayloadAction } from "@reduxjs/toolkit";
import type { RootState } from "./store";

// Define a type for the slice state
interface followState {
  following: string[];
}

// Define the initial state using that type
const initialState: followState = {
  following: [],
};

export const followSlice = createSlice({
  name: "follow",
  // `createSlice` will infer the state type from the `initialState` argument
  initialState,
  reducers: {
    // Use the PayloadAction type to declare the contents of `action.payload`
    setInitialFollowing: (state, action: PayloadAction<string[]>) => {
      state.following = action.payload;
    },
    addToFollowing: (state, action: PayloadAction<string>) => {
      const userId = action.payload;
      state.following = [...state.following, userId];
    },
    removeFromFollowing: (state, action: PayloadAction<string>) => {
      const index = state.following.indexOf(action.payload);
      // only splice array when item is found
      if (index > -1) {
        state.following.splice(index, 1); // 2nd parameter means remove one item only
      }
    },
  },
});

export const { setInitialFollowing, addToFollowing, removeFromFollowing } =
  followSlice.actions;

// Other code such as selectors can use the imported `RootState` type
export const selectFollowing = (state: RootState) => state.follow.following;

export default followSlice.reducer;
