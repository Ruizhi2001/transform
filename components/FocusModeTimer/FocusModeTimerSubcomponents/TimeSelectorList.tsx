import { ScrollView, StyleSheet, Text } from "react-native";
import React from "react";
import { TimerSelectorButton } from "./TimerSelectorButton";

type Props = {
  toggleTimer: (duration: number) => void;
};

export const TimeSelectorList: React.FC<Props> = ({ toggleTimer }) => {
  const durations = [20, 30, 40, 50, 60, 70, 80, 90, 100, 110, 120];

  return (
    <ScrollView
      horizontal={true}
      showsHorizontalScrollIndicator={false}
      // style={styles.scrollview}
    >
      {durations.map((duration) => (
        <TimerSelectorButton
          key={duration}
          duration={duration}
          toggleTimer={() => toggleTimer(duration)}
        />
      ))}
    </ScrollView>
  );
};

const styles = StyleSheet.create({
  scrollview: {
    flex: 1,
  },
});
