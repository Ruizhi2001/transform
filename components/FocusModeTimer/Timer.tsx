import { StyleSheet, Text, View } from "react-native";
import React, { useEffect, useState } from "react";
import { TimerToggleButton } from "./FocusModeTimerSubcomponents/TimerToggleButton";
import { TimerCountDownDisplay } from "./FocusModeTimerSubcomponents/TimerCountDownDisplay";
import { TimeSelectorList } from "./FocusModeTimerSubcomponents/TimeSelectorList";
import {
  DocumentReference,
  doc,
  getDoc,
  increment,
  updateDoc,
} from "firebase/firestore";
import { auth, db } from "../../FirebaseConfig";
import { useAppDispatch } from "../../redux/hooks";
import { incrementByAmount } from "../../redux/expSlice";

const defaultDuration = 20 * 60 * 1000;

export const Timer: React.FC = () => {
  const [timerCount, setTimerCount] = useState<number>(defaultDuration);
  const [timerInterval, setTimerInterval] = useState<NodeJS.Timer | null>(null);
  const [isTimerRunning, setIsTimerRunning] = useState<boolean>(false);
  const user = auth.currentUser;

  let userDocRef: DocumentReference;
  const dispatch = useAppDispatch();

  if (user) {
    userDocRef = doc(db, "users", user.uid);
    getDoc(userDocRef);
  }

  const startTimer = () => {
    const id = setInterval(() => setTimerCount((prev) => prev - 1000), 1000);
    setIsTimerRunning(true);
    setTimerInterval(id);
  };

  const stopTimer = () => {
    setIsTimerRunning(false);
    if (timerInterval != null) {
      clearInterval(timerInterval);
      setTimerCount(defaultDuration);
    }
  };

  const setTimer = (duration: number) => {
    setTimerCount(duration * 60 * 1000);
  };

  useEffect(() => {
    if (timerCount === 0) {
      updateDoc(userDocRef, {
        exp: increment(timerCount),
      });
      dispatch(incrementByAmount(timerCount));
      stopTimer();
    }
  }, [timerCount]);

  return (
    <View style={styles.container}>
      <Text style={styles.timerText}>
        {isTimerRunning ? "TRANSFORMING!!!" : "TRANSFORM"}
      </Text>
      <TimerCountDownDisplay timerDate={new Date(timerCount)} />
      {isTimerRunning ? null : <TimeSelectorList toggleTimer={setTimer} />}
      <TimerToggleButton
        isTimerRunning={isTimerRunning}
        startTimer={startTimer}
        stopTimer={stopTimer}
      />
    </View>
  );
};

const styles = StyleSheet.create({
  container: {
    alignItems: "center",
    justifyContent: "space-between",
  },
  timerText: {
    fontSize: 30,
    alignItems: "center",
    fontWeight: "700",
  },
});
