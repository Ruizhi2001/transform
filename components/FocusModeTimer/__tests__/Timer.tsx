import React from "react";
import renderer, { ReactTestInstance, act } from "react-test-renderer";
import { Timer, defaultDuration } from "../Timer";
import { updateDoc, increment } from "firebase/firestore";
import expReducer, { incrementByAmount } from "../../../redux/expSlice";
import { Text } from "react-native";
import { TimerToggleButton } from "../FocusModeTimerSubcomponents/TimerToggleButton";
import { configureStore } from "@reduxjs/toolkit";
import { Provider } from "react-redux";
import { useAppDispatch, useAppSelector } from "../../../redux/hooks";

// Mock the useAppDispatch and auth.currentUser hook
jest.mock("../../../redux/hooks", () => ({
  useAppDispatch: jest.fn(() => jest.fn),
}));

describe("Timer", () => {
  it("should match the snapshot", () => {
    const tree = renderer.create(<Timer />).toJSON();
    expect(tree).toMatchSnapshot();
  });

  it("should start and stop the timer correctly", () => {
    jest.useFakeTimers();
    const tree = renderer.create(<Timer />);
    const instance = tree.root as ReactTestInstance;

    // Verify that the timer is initially not running
    const allTextComponents = instance.findAllByType(
      Text
    ) as ReactTestInstance[];
    const timerText = allTextComponents.find(
      (text) => text.props.children === "TRANSFORM"
    );
    expect(timerText?.props.children).toEqual("TRANSFORM");

    // Start the timer
    const timerToggleButton = instance.findByType(
      TimerToggleButton
    ) as ReactTestInstance;
    act(() => {
      timerToggleButton.props.startTimer(); // Simulate starting the timer
    });
    expect(timerText?.props.children).toEqual("TRANSFORMING!!!");

    // Fast-forward the timer by 5 seconds (for testing purposes)
    act(() => {
      jest.advanceTimersByTime(5000);
    });

    // Verify that the timer is still running
    expect(timerText?.props.children).toEqual("TRANSFORMING!!!");

    // Stop the timer
    act(() => {
      timerToggleButton.props.stopTimer(); // Simulate stopping the timer
    });
    expect(timerText?.props.children).toEqual("TRANSFORM");

    // Fast-forward the timer by another 5 seconds (for testing purposes)
    act(() => {
      jest.advanceTimersByTime(5000);
    });

    // Verify that the timer is still stopped
    expect(timerText?.props.children).toEqual("TRANSFORM");
  });

  // it("should update the timerCount and dispatch actions when the timer reaches 0", () => {
  //   jest.useFakeTimers();

  //   // Create a real Redux store with your root reducer
  //   const store = configureStore({
  //     reducer: {
  //       exp: expReducer,
  //     },
  //   });

  //   const tree = renderer.create(
  //     <Provider store={store}>
  //       <Timer />
  //     </Provider>
  //   );

  //   const instance = tree.root as ReactTestInstance;
  //   const timerToggleButton = instance.findByType(
  //     TimerToggleButton
  //   ) as ReactTestInstance;

  //   const initialState = store.getState().exp.value;

  //   // Start the timer
  //   act(() => {
  //     timerToggleButton.props.startTimer();
  //   });

  //   act(() => {
  //     jest.advanceTimersByTime(defaultDuration);
  //   });

  //   act(() => {
  //     useAppDispatch()(incrementByAmount(defaultDuration / 1000));
  //   });

  //   const { value } = useAppSelector((store) => store.exp);

  //   expect(value).toEqual(initialState + defaultDuration / 1000);

  //   // Verify that the firestore updateDoc function is called with the correct argument
  //   expect(updateDoc).toHaveBeenCalledWith(expect.any(Object), {
  //     exp: increment(defaultDuration),
  //   });
  // });
});
