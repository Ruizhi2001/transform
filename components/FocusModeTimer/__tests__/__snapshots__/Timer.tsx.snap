// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Timer should match the snapshot 1`] = `
<View
  style={
    {
      "alignItems": "center",
      "justifyContent": "space-between",
    }
  }
>
  <Text
    style={
      {
        "alignItems": "center",
        "fontSize": 30,
        "fontWeight": "700",
      }
    }
  >
    TRANSFORM
  </Text>
  <View>
    <Text
      style={
        {
          "fontSize": 90,
          "fontWeight": "700",
        }
      }
    >
      00
      :
      20
      :
      00
    </Text>
  </View>
  <RCTScrollView
    horizontal={true}
    showsHorizontalScrollIndicator={false}
  >
    <View>
      <View>
        <View
          accessibilityState={
            {
              "busy": undefined,
              "checked": undefined,
              "disabled": undefined,
              "expanded": undefined,
              "selected": undefined,
            }
          }
          accessibilityValue={
            {
              "max": undefined,
              "min": undefined,
              "now": undefined,
              "text": undefined,
            }
          }
          accessible={true}
          collapsable={false}
          focusable={true}
          onClick={[Function]}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            {
              "alignItems": "center",
              "borderColor": "green",
              "borderRadius": 10,
              "borderWidth": 6,
              "margin": 10,
              "opacity": 1,
              "padding": 10,
            }
          }
        >
          <Text
            style={
              {
                "color": "black",
              }
            }
          >
            20
          </Text>
        </View>
      </View>
      <View>
        <View
          accessibilityState={
            {
              "busy": undefined,
              "checked": undefined,
              "disabled": undefined,
              "expanded": undefined,
              "selected": undefined,
            }
          }
          accessibilityValue={
            {
              "max": undefined,
              "min": undefined,
              "now": undefined,
              "text": undefined,
            }
          }
          accessible={true}
          collapsable={false}
          focusable={true}
          onClick={[Function]}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            {
              "alignItems": "center",
              "borderColor": "green",
              "borderRadius": 10,
              "borderWidth": 6,
              "margin": 10,
              "opacity": 1,
              "padding": 10,
            }
          }
        >
          <Text
            style={
              {
                "color": "black",
              }
            }
          >
            30
          </Text>
        </View>
      </View>
      <View>
        <View
          accessibilityState={
            {
              "busy": undefined,
              "checked": undefined,
              "disabled": undefined,
              "expanded": undefined,
              "selected": undefined,
            }
          }
          accessibilityValue={
            {
              "max": undefined,
              "min": undefined,
              "now": undefined,
              "text": undefined,
            }
          }
          accessible={true}
          collapsable={false}
          focusable={true}
          onClick={[Function]}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            {
              "alignItems": "center",
              "borderColor": "green",
              "borderRadius": 10,
              "borderWidth": 6,
              "margin": 10,
              "opacity": 1,
              "padding": 10,
            }
          }
        >
          <Text
            style={
              {
                "color": "black",
              }
            }
          >
            40
          </Text>
        </View>
      </View>
      <View>
        <View
          accessibilityState={
            {
              "busy": undefined,
              "checked": undefined,
              "disabled": undefined,
              "expanded": undefined,
              "selected": undefined,
            }
          }
          accessibilityValue={
            {
              "max": undefined,
              "min": undefined,
              "now": undefined,
              "text": undefined,
            }
          }
          accessible={true}
          collapsable={false}
          focusable={true}
          onClick={[Function]}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            {
              "alignItems": "center",
              "borderColor": "green",
              "borderRadius": 10,
              "borderWidth": 6,
              "margin": 10,
              "opacity": 1,
              "padding": 10,
            }
          }
        >
          <Text
            style={
              {
                "color": "black",
              }
            }
          >
            50
          </Text>
        </View>
      </View>
      <View>
        <View
          accessibilityState={
            {
              "busy": undefined,
              "checked": undefined,
              "disabled": undefined,
              "expanded": undefined,
              "selected": undefined,
            }
          }
          accessibilityValue={
            {
              "max": undefined,
              "min": undefined,
              "now": undefined,
              "text": undefined,
            }
          }
          accessible={true}
          collapsable={false}
          focusable={true}
          onClick={[Function]}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            {
              "alignItems": "center",
              "borderColor": "green",
              "borderRadius": 10,
              "borderWidth": 6,
              "margin": 10,
              "opacity": 1,
              "padding": 10,
            }
          }
        >
          <Text
            style={
              {
                "color": "black",
              }
            }
          >
            60
          </Text>
        </View>
      </View>
      <View>
        <View
          accessibilityState={
            {
              "busy": undefined,
              "checked": undefined,
              "disabled": undefined,
              "expanded": undefined,
              "selected": undefined,
            }
          }
          accessibilityValue={
            {
              "max": undefined,
              "min": undefined,
              "now": undefined,
              "text": undefined,
            }
          }
          accessible={true}
          collapsable={false}
          focusable={true}
          onClick={[Function]}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            {
              "alignItems": "center",
              "borderColor": "green",
              "borderRadius": 10,
              "borderWidth": 6,
              "margin": 10,
              "opacity": 1,
              "padding": 10,
            }
          }
        >
          <Text
            style={
              {
                "color": "black",
              }
            }
          >
            70
          </Text>
        </View>
      </View>
      <View>
        <View
          accessibilityState={
            {
              "busy": undefined,
              "checked": undefined,
              "disabled": undefined,
              "expanded": undefined,
              "selected": undefined,
            }
          }
          accessibilityValue={
            {
              "max": undefined,
              "min": undefined,
              "now": undefined,
              "text": undefined,
            }
          }
          accessible={true}
          collapsable={false}
          focusable={true}
          onClick={[Function]}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            {
              "alignItems": "center",
              "borderColor": "green",
              "borderRadius": 10,
              "borderWidth": 6,
              "margin": 10,
              "opacity": 1,
              "padding": 10,
            }
          }
        >
          <Text
            style={
              {
                "color": "black",
              }
            }
          >
            80
          </Text>
        </View>
      </View>
      <View>
        <View
          accessibilityState={
            {
              "busy": undefined,
              "checked": undefined,
              "disabled": undefined,
              "expanded": undefined,
              "selected": undefined,
            }
          }
          accessibilityValue={
            {
              "max": undefined,
              "min": undefined,
              "now": undefined,
              "text": undefined,
            }
          }
          accessible={true}
          collapsable={false}
          focusable={true}
          onClick={[Function]}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            {
              "alignItems": "center",
              "borderColor": "green",
              "borderRadius": 10,
              "borderWidth": 6,
              "margin": 10,
              "opacity": 1,
              "padding": 10,
            }
          }
        >
          <Text
            style={
              {
                "color": "black",
              }
            }
          >
            90
          </Text>
        </View>
      </View>
      <View>
        <View
          accessibilityState={
            {
              "busy": undefined,
              "checked": undefined,
              "disabled": undefined,
              "expanded": undefined,
              "selected": undefined,
            }
          }
          accessibilityValue={
            {
              "max": undefined,
              "min": undefined,
              "now": undefined,
              "text": undefined,
            }
          }
          accessible={true}
          collapsable={false}
          focusable={true}
          onClick={[Function]}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            {
              "alignItems": "center",
              "borderColor": "green",
              "borderRadius": 10,
              "borderWidth": 6,
              "margin": 10,
              "opacity": 1,
              "padding": 10,
            }
          }
        >
          <Text
            style={
              {
                "color": "black",
              }
            }
          >
            100
          </Text>
        </View>
      </View>
      <View>
        <View
          accessibilityState={
            {
              "busy": undefined,
              "checked": undefined,
              "disabled": undefined,
              "expanded": undefined,
              "selected": undefined,
            }
          }
          accessibilityValue={
            {
              "max": undefined,
              "min": undefined,
              "now": undefined,
              "text": undefined,
            }
          }
          accessible={true}
          collapsable={false}
          focusable={true}
          onClick={[Function]}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            {
              "alignItems": "center",
              "borderColor": "green",
              "borderRadius": 10,
              "borderWidth": 6,
              "margin": 10,
              "opacity": 1,
              "padding": 10,
            }
          }
        >
          <Text
            style={
              {
                "color": "black",
              }
            }
          >
            110
          </Text>
        </View>
      </View>
      <View>
        <View
          accessibilityState={
            {
              "busy": undefined,
              "checked": undefined,
              "disabled": undefined,
              "expanded": undefined,
              "selected": undefined,
            }
          }
          accessibilityValue={
            {
              "max": undefined,
              "min": undefined,
              "now": undefined,
              "text": undefined,
            }
          }
          accessible={true}
          collapsable={false}
          focusable={true}
          onClick={[Function]}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            {
              "alignItems": "center",
              "borderColor": "green",
              "borderRadius": 10,
              "borderWidth": 6,
              "margin": 10,
              "opacity": 1,
              "padding": 10,
            }
          }
        >
          <Text
            style={
              {
                "color": "black",
              }
            }
          >
            120
          </Text>
        </View>
      </View>
    </View>
  </RCTScrollView>
  <View
    accessibilityState={
      {
        "busy": undefined,
        "checked": undefined,
        "disabled": undefined,
        "expanded": undefined,
        "selected": undefined,
      }
    }
    accessibilityValue={
      {
        "max": undefined,
        "min": undefined,
        "now": undefined,
        "text": undefined,
      }
    }
    accessible={true}
    collapsable={false}
    focusable={true}
    onBlur={[Function]}
    onClick={[Function]}
    onFocus={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      {
        "alignItems": "center",
        "borderColor": "gold",
        "borderRadius": 10,
        "borderWidth": 3,
      }
    }
  >
    <View>
      <Text
        style={
          {
            "fontSize": 30,
            "fontWeight": "700",
          }
        }
      >
        ⚡TRANSFORM NOW!⚡
      </Text>
    </View>
  </View>
</View>
`;
